build.sh

```diff
@@ -21,7 +21,7 @@ if test "$*" = "--none"; then
 elif test "$*" = "--common"; then
     cmake . -DUSE_GSL=1 -DUSE_PCAP=1 -DUSE_SSL= -DUSE_SCTP=
 elif test "$*" = "--full"; then
-    cmake . -DUSE_GSL=1 -DUSE_PCAP=1 -DUSE_SSL=1 -DUSE_SCTP=1
+    cmake . -DUSE_GSL=1 -DUSE_PCAP=1 -DUSE_SSL=1 -DUSE_SCTP=1 -DUSE_SHA256=1
```

Comment:
Actually why do we need this flag at all? Is there a reason *not* to support it when USE_SSL is set?

Reply:
I don't think it's necessary to compile SHA-256 by default because it will always require openssl.

At first I also thought about turning on SHA-256 if `USE_SSL` flag is set, but then I abandoned this idea because these are unrelated things. So I added an extra flag for flexibility.

So, do you think we should enable algorithm by default, or only when `USE_SSL` flag is set?

Reply:
I think `USE_SSL` is enough (without it, you can't assume this machine has openssl).

Reply:
Done.